py/selenium/webdriver/common/by.py

```diff
@@ -19,17 +19,20 @@
 The By implementation.
 """
 
+from enum import Enum, unique, auto
 
-class By:
+
+@unique
+class By(Enum):
     """
     Set of supported locator strategies.
     """
 
-    ID = "id"
-    XPATH = "xpath"
-    LINK_TEXT = "link text"
-    PARTIAL_LINK_TEXT = "partial link text"
-    NAME = "name"
-    TAG_NAME = "tag name"
-    CLASS_NAME = "class name"
-    CSS_SELECTOR = "css selector"
+    ID = auto()
```

Comment:
won't this throw `using:` way off?

Reply:
> won't this throw `using:` way off?

I'm afraid but could you explain in detail? Do you mean that constants are used in somewhere?

Reply:
yes, the old `str` value is passed into the `using: by`